_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,156 @@
+
+What you see today when you open the [Artsy iOS app](https://itunes.apple.com/us/app/artsy-collect-and-bid-on-fine-art-design/id703796080?mt=8) is a mix of Objective-C, Swift and React Native. Objective-C and Swift continue to provide a lot of over-arching cross-View Controller code. While individual representations of Artsy resources tend to be built in React Native. All of our React Native code uses Relay to handle API integration. This [code is open-source](https://github.com/artsy/eigen).
+
+You can also find Artsy on [Alexa](http://alexa.atsy.net) and [Google Home](http://assistant.artsy.net), which are both open-source Node.js applications. There is also an open-source [Apple TV](https://github.com/artsy/emergence/) app built in Swift.
+
+Our core API serves the public facets of our product, many of our own internal applications, and even [some of your own projects](https://developers.artsy.net). It's built with [Ruby](https://www.ruby-lang.org/en), [Rack](http://rack.github.io), [Rails](http://rubyonrails.org), and [Grape](https://github.com/intridea/grape) serving primarily JSON. The API is hosted on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Memcached](http://memcached.org) for caching and [Redis](https://redis.io) for background queues. It runs background jobs with [delayed_job](https://github.com/collectiveidea/delayed_job). We used to employ [Apache Solr](http://lucene.apache.org/solr) and even [Google Custom Search](https://www.google.com/cse) for the many search functions, but have since consolidated on [Elasticsearch](https://www.elastic.co).
+
+Most modern code for both the website and the iOS app use an orchestration layer which is powered by [GraphQL](http://graphql.org). Our GraphQL server is an [Express](http://expressjs.com) app, using [express-graphql](https://github.com/graphql/express-graphql) to provide a single API end-point. The API does not access our data directly, but forwards requests to the core API or other services. We have been migrating shared display logic into the GraphQL server, to make it easier to build consistent clients. This [code is open-source](https://github.com/artsy/metaphysics).
+
+Consistently, our front-end code [has moved towards](/blog/2017/02/05/Front-end-JavaScript-at-Artsy-2017/) using React across all platforms along with introducing stricter JavaScript languages like TypeScript over CoffeeScript in order to provide better tooling.
```

Comment:
It might bear mentioning that ES6 has largely obviated the advantages we saw in using CoffeeScript, and that we're beginning to use Typescript on top of that.